src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -242,7 +242,7 @@ protected function buildDeployment(callable $getFunctionsDevice, Func $queueForF
                     Console::execute('mkdir -p ' . $tmpDirectory . '/' . $rootDirectory, '', $stdout, $stderr);
 
                     // Merge template into user repo
-                    Console::execute('cp -rfn ' . $tmpTemplateDirectory . '/' . $templateRootDirectory . '/* ' . $tmpDirectory . '/' . $rootDirectory, '', $stdout, $stderr);
+                    Console::execute('rsync -av --exclude \'.git\' ' . $tmpTemplateDirectory . '/' . $templateRootDirectory . '/ ' . $tmpDirectory . '/' . $rootDirectory, '', $stdout, $stderr);
```

Comment:
We have to use rsync to copy all files, including hidden, but exclude .git else there is problems related to sparse checkout config set by the clone command.